src/Model/Map.php

```diff
@@ -2,15 +2,216 @@
-{
+class Map extends CustomPost {
+
+	/**
+	 * get geo data from location metadata
+	 *
+	 * @param $mapItemTerms array of term ids
+	 *
+	 * @return array with postIDs as keys for an array with location data relevant for the map
+	 * @throws Exception
+	 */
+	public function get_locations( array $mapItemTerms ): array {
+		$locations = [];
+
+		$show_location_contact = $this->getMeta( 'show_location_contact' );
+
+		$preset_categories          = $this->getMeta( 'cb_items_preset_categories' );
+		$preset_location_categories = $this->getMeta( 'cb_locations_preset_categories' );
+
+
+		$args = [
+			'post_type'      => Location::$postType,
+			'posts_per_page' => - 1,
+			'post_status'    => 'publish',
+			'meta_query'     => [
+				[
+					'key'          => 'geo_longitude',
+					'meta_compare' => 'EXISTS',
+				],
+			],
+		];
+
+		$locationObjects = \CommonsBooking\Repository\Location::get(
+			$args,
+			true
+		);
+
+		/** @var \CommonsBooking\Model\Location $post */
+		foreach ( $locationObjects as $post ) {
+			$location_meta = get_post_meta( $post->ID, null, true );
+
+			//set serialized empty array if not set
+			$closed_days = isset( $location_meta['commons-booking_location_closeddays'] ) ? $location_meta['commons-booking_location_closeddays'][0] : 'a:0:{}';
+
+			$items = [];
+
+			/**
+			 * filters out not preset location categories, if location categories are set
+			 */
+
+			if ( $preset_location_categories ) {
+				if ( ! has_term( $preset_location_categories, 'cb_locations_category', $post->ID ) ) {
+					continue; //skip to next location in loop
+				}
+			}
+
+			foreach ( Item::getByLocation( $post->ID, true ) as $item ) {
+
+				$item_terms = wp_get_post_terms(
+					$item->ID,
+					\CommonsBooking\Wordpress\CustomPostType\Item::$postType . 's_category'
+				);
+				if ( is_array( $item_terms ) && count( $item_terms ) ) {
+					$item_terms = array_map(
+						function ( $item ) {
+							return $item->term_id;
+						},
+						$item_terms
+					);
+				}
+
+				/**
+				 * If current item has a category, that isn't in map config, we'll skip it.
+				 */
+				if ( count( $mapItemTerms ) && count( $item_terms ) && ! count( array_intersect( $item_terms, $mapItemTerms ) ) ) {
+					continue;
+				}
+
+				/**
+				 * Filter items by preset item categories
+				 */
+
+				if ( $preset_categories ) {
+					//check if preset category is in items
+					if ( ! has_term( $preset_categories, 'cb_items_category', $item->ID ) ) {
+						continue; //skip to next item in loop
+					}
+				}
+
+
+				$timeframesData = [];
+				$timeframes     = Timeframe::getBookableForCurrentUser(
+					[ $post->ID ],
+					[ $item->ID ],
+					null,
+					true
+				);
+
+				/** @var \CommonsBooking\Model\Timeframe $timeframe */
+				foreach ( $timeframes as $timeframe ) {
+					$startDate        = date( 'Y-m-d', $timeframe->getStartDate() );
+					$endDate          = $timeframe->getEndDate() ?: date( 'Y-m-d', strtotime( '2999-01-01' ) );
```

Comment:
Frage: Das machen wir bisher nicht so oder? Also mit dem EndDate in der Zukunft, wenn es null ist.

Reply:
Also finde ich grundsätzlich gut. Habe es bisher nur noch nie gesehen im Code.